consensus/types/src/beacon_state/tree_hash_cache.rs

```diff
@@ -21,9 +22,66 @@ const NODES_PER_VALIDATOR: usize = 15;
 /// Do not set to 0.
 const VALIDATORS_PER_ARENA: usize = 4_096;
 
+#[derive(Debug, PartialEq, Clone, Encode, Decode)]
+pub struct Eth1DataVotesTreeHashCache<T: EthSpec> {
+    arena: CacheArena,
+    tree_hash_cache: TreeHashCache,
+    voting_period: u64,
+    roots: VariableList<Hash256, T::SlotsPerEth1VotingPeriod>,
+}
+
+impl<T: EthSpec> Eth1DataVotesTreeHashCache<T> {
+    /// Instantiates a new cache.
+    ///
+    /// Allocates the necessary memory to store all of the cached Merkle trees but does perform any
```

Comment:
Perhaps this comment should clarify that it doesn't hash any internal nodes of the tree (the leaves are hashed)